esmvalcore/_main.py

```diff
@@ -47,6 +48,22 @@
 """ + __doc__
 
 
+def parse_resume(resume, recipe):
+    """Set `resume` to a correct value and sanity check."""
+    if not resume:
+        resume = []
+    if isinstance(resume, str):
+        resume = [resume]
+    # Sanity check resume directories:
+    current_recipe = recipe.read_text()
+    for resume_dir in resume:
+        resume_recipe = Path(resume_dir, 'run', recipe.name)
+        if current_recipe != resume_recipe.read_text():
+            raise ValueError(f'Only identical recipes can be resumed, but '
+                             f'{resume_recipe} is different from {recipe}')
```

Comment:
I would argue that it's enough to have the same `preprocessors` section of the recipe rather than the full recipe - if a user decides to change variables/input data parameters that are not covered in the preprocessing, then the run will fail and that's the responsibility of the user; it would, however, be nice to have the same preprocessed output run with different diagnostics

Reply:
actually - not even the same `preprocessors` but rather, a subset of those, eg if the user wants to add a `multimodel` preproc at the bottom, that should be fine, and have to regenerate all the previous setps. Of course, this is not trivial - what can we allow and what we can't?

Reply:
I agree these would all be very nice improvements, but I would like to try to finish this before the feature freeze next week. I believe that this feature is already useful with the current simple check.

Reply:
yes, agreed - just my two cents for a future improvement, am not gonna hold this going through :motorway: